gc3apps/bf.uzh.ch/gcsvBulk.py

```diff
@@ -0,0 +1,216 @@
+#   modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+
+It uses the generic `gc3libs.cmdline.SessionBasedScript` framework.
+
+See the output of ``gcsvBulk.py --help`` for program usage
+instructions.
+
+Input parameters consists of:
+
+...
+
+Options:
+"""
+
+__version__ = 'development version (SVN $Revision$)'
+# summary of user-visible changes
+__changelog__ = """
+  2016-05-11:
+  * Initial version
+"""
+__author__ = 'Adrian Etter <adrian.etter@uzh.ch>'
+__docformat__ = 'reStructuredText'
+
+
+# run script, but allow GC3Pie persistence module to access classes defined here;
+# for details, see: https://github.com/uzh/gc3pie/issues/95
+if __name__ == "__main__":
+    import gcsvBulk
+    gcsvBulk.GcsvBulkScript().run()
+
+from argparse import ArgumentError
+import os
+from os.path import basename, exists, join, realpath
+import sys
+import time
+import tempfile
+
+from math import ceil
+
+import shutil
+# import csv
+
+from pkg_resources import Requirement, resource_filename
+
+import gc3libs
+import gc3libs.exceptions
+from gc3libs import Application, Run, Task
+from gc3libs.cmdline import SessionBasedScript, executable_file
+import gc3libs.utils
+from gc3libs.quantity import Memory, kB, MB, MiB, GB, Duration, hours, minutes, seconds
+from gc3libs.workflow import RetryableTask
+
+DEFAULT_CORES = 4
+DEFAULT_MEMORY = Memory(7,GB)
+DEFAULT_WALLTIME = Duration(200,hours)
+
+## custom application class
+class GcsvBulkApplication(Application):
+    """
+    Custom class to wrap the execution of a matlab script with a csv as input.
+    A convenient wrapper script is provided to facilitate how the command file
+    is executed on the remote end.
+    """
+    application_name = 'gcsvBulk'
+
+    def __init__(self, input_file, input_function, input_csvfile, limit, skip, **extra_args):
+
+        self.output_dir = extra_args['output_dir']
+        self.input_function = input_function
+        #
+        inputs = dict()
+
+        gcsvBulk_wrapper_m = resource_filename(Requirement.parse("gc3pie"),
+                                              "gc3libs/etc/gcsvBulk_wrapper.m")
+        #
+        inputs[gcsvBulk_wrapper_m] = os.path.basename(gcsvBulk_wrapper_m)
+        # inputs['foo'] = 'foo'
+        #
+        inputs[input_file] = "%s" % os.path.basename(input_file)
+        inputs[input_csvfile] = "%s" % os.path.basename(input_csvfile)
```

Comment:
No point in converting to string something that is already a string. (Applies to the two lines above.)